### PR TITLE
Fix EDT violations and chat refresh

### PR DESCRIPTION
**Pull request by pkukielka**
## Test plan

Full QA before today's release.

### CODE REVIEW

**Review by Sa1to — APPROVED** [review]
LGTM

**Review by mkondratek — APPROVED** [review]
lgtm - quickly tested, no slow ops reported

File: src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt [hunk 1]
```diff
@@ -21,7 +21,9 @@ class HistoryService : SimplePersistentStateComponent<HistoryState>(HistoryState
   fun update(internalId: String, chatMessages: List<ChatMessage>) {
     val found = getChatOrCreate(internalId)
     found.messages = chatMessages.map(::convertToMessageState).toMutableList()
-    found.setUpdatedTimeAt(LocalDateTime.now())
+    if (chatMessages.lastOrNull()?.speaker == Speaker.HUMAN) {
+      found.setUpdatedTimeAt(LocalDateTime.now())
+    }
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt:26]: This fixes the flickering list when +2 responses are being generated.

File: src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatusService.kt [hunk 2]
```diff
@@ -75,18 +76,12 @@ class CodyAutocompleteStatusService : CodyAutocompleteStatusListener, Disposable
   }
 
   private fun updateCodyStatusBarIcons() {
-    val action = Runnable {
+    UIUtil.invokeLaterIfNeeded {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatusService.kt:79]: unrel fix

**Review by mkondratek — APPROVED** [review]

**Review by Sa1to — APPROVED** [review]
LGTM

**Review by exigow — APPROVED** [review]
Works on Windows!

**Pull request merged**